### PR TITLE
[3006.x] saltutil.cmd: "Authentication error occurred." when master is not running as root

### PR DESCRIPTION
**Pull request by dmurphy18**
### What does this PR do?
Allows for multiple keys presented for authentication rather than just a single key for root, now that we can have the root key and the salt key presented

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/64398

### Previous Behavior
Would return 'Authentication error occurred.'

### New Behavior
Allows for salt key presented, and operations succeed, in addition to the root key.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [X] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: tests/pytests/unit/daemons/masterapi/test_local_funcs.py [hunk 1]
```diff
@@ -1,21 +1,36 @@
...
+log = logging.getLogger(__name__)
+
 pytestmark = [
     pytest.mark.slow_test,
 ]
 
+test_check_key = {
+    "test": "mGXdurU1c8lXt5cmpbGq4rWvrOvDXxkwI9gbkP5CBBjpyGWuB8vkgz9r+sjjG0wVDL9/uFuREtk=",
+    "root": "2t5HHv/ek2wIFh8tTX2c3hdt+6V+93xKlcXb7IlGLIszOeCVv2NuH38LyCw9UwQTfUFTeseXhSs=",
+}
```
Comment (s0undt3ch) [inline tests/pytests/unit/daemons/masterapi/test_local_funcs.py:19]:
```suggestion
@pytest.fixture
def check_keys():
    return {
    "test": "mGXdurU1c8lXt5cmpbGq4rWvrOvDXxkwI9gbkP5CBBjpyGWuB8vkgz9r+sjjG0wVDL9/uFuREtk=",
    "root": "2t5HHv/ek2wIFh8tTX2c3hdt+6V+93xKlcXb7IlGLIszOeCVv2NuH38LyCw9UwQTfUFTeseXhSs=",
}
```

File: tests/pytests/unit/daemons/masterapi/test_local_funcs.py [hunk 2]
```diff
@@ -1,21 +1,36 @@
...
+
 
 @pytest.fixture
 def local_funcs():
     opts = salt.config.master_config(None)
     return masterapi.LocalFuncs(opts, "test-key")
```
Comment (s0undt3ch) [inline tests/pytests/unit/daemons/masterapi/test_local_funcs.py:25]:
```suggestion
def local_funcs(master_opts):
    return masterapi.LocalFuncs(master_opts, "test-key")
```

File: tests/pytests/unit/daemons/masterapi/test_local_funcs.py [hunk 3]
```diff
@@ -1,21 +1,36 @@
...
 
 
+@pytest.fixture
+def check_local_funcs():
+    opts = salt.config.master_config(None)
+    return masterapi.LocalFuncs(opts, test_check_key)
```
Comment (s0undt3ch) [inline tests/pytests/unit/daemons/masterapi/test_local_funcs.py:31]:
```suggestion
def check_local_funcs(master_opts, check_keys):
    return masterapi.LocalFuncs(master_opts, check_keys)
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]
* `master_opts` fixture is tailored for not requiring root user to run tests, on top of other small config changes
* let's not define top level mutable objects, use a fixture.

**Review by twangboy — DISMISSED** [review]

File: tests/pytests/unit/daemons/masterapi/test_local_funcs.py (hunk 2 above)
Comment (dmurphy18) [reply tests/pytests/unit/daemons/masterapi/test_local_funcs.py]: done

File: tests/pytests/unit/daemons/masterapi/test_local_funcs.py (hunk 1 above)
Comment (dmurphy18) [reply tests/pytests/unit/daemons/masterapi/test_local_funcs.py]: done

File: tests/pytests/unit/daemons/masterapi/test_local_funcs.py (hunk 3 above)
Comment (dmurphy18) [reply tests/pytests/unit/daemons/masterapi/test_local_funcs.py]: done

File: salt/auth/__init__.py [hunk 4]
```diff
@@ -331,28 +332,36 @@ def authenticate_key(self, load, key):
...
+                    "Master runs as %r, but user in payload is %r",
+                    self.opts.get("user", "root"),
+                    load["user"],
+                )
+                log.warning(error_msg)
+                return False
+
             elif auth_user.is_running_user():
                 if auth_key != key.get(load["user"]):
+                    load_user = load["user"]
```
Comment (s0undt3ch) [inline salt/auth/__init__.py:357]: Where is this `load_user` variable used?

**Review by s0undt3ch — DISMISSED** [review]

File: salt/auth/__init__.py (hunk 4 above)
Comment (dmurphy18) [reply salt/auth/__init__.py:357]: Missed line left over from debugging statements now removed, will remove it
Thanks for catching

**Review by whytewolf — APPROVED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**